src/Sulu/Bundle/AdminBundle/Resources/js/views/Datagrid/Datagrid.js

```diff
@@ -93,6 +101,25 @@ class Datagrid extends React.Component<ViewProps> {
         router.bind('limit', this.datagridStore.limit, 10);
     }
 
+    buildDatagridStoreOptions(
+        apiOptions: Object,
+        attributes: Object,
+        routerAttributesToDatagridStore: { [string | number]: string }
+    ) {
+        const datagridStoreOptions = apiOptions ? apiOptions : {};
+
+        Object.keys(routerAttributesToDatagridStore).forEach((key) => {
+            const attributeName = routerAttributesToDatagridStore[key];
+            if (!isNaN(key)) {
+                key = routerAttributesToDatagridStore[key];
```

Comment:
overriding this variable means the initial value is lost, which can lead to confusing situations... Can`t we inline that in the index acces below like `datagridStoreOptions[isNaN(key) ? key : routerAttributesToDatagridStore[key]]`? Or if that seems to unclear to you, maybe at least remove the `if` like this: `const datagridOptionKey = isNaN(key) ? key : ...`?